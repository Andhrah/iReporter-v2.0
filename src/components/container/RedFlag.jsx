/* eslint-disable react/no-unused-state */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import RedFlagView from '../presentational/RedFlag';
import { getSingleRedflag } from '../../actions/singleRedFlagAction';
import Navbar from '../presentational/Navbar';
import checkAuthentication from '../../utils/authentication';

export class RedFlag extends Component {
  state = {
    editMode: false,
    activeEditField: '',
    showDeleteModal: false,
    deleted: false
  };

  componentDidMount() {
    const { match: { params: { id } }, getSingleRedflag: fetchSingleRedflag } = this.props;
    fetchSingleRedflag(id);
  }

  render() {
    const { singleRedFlag: { data, loading } } = this.props;
    checkAuthentication();
    return (
      !loading
      && (
      <>
        <Navbar
          firstname={localStorage.getItem('firstname')}
          lastname={localStorage.getItem('lastname')}
          email={localStorage.getItem('email')}
          username={localStorage.getItem('username')}
          />
        <RedFlagView redFlag={data[0]} />
      </>
      )
    );
  }
}

RedFlag.propTypes = {
  match: PropTypes.object.isRequired,
  getSingleRedflag: PropTypes.func.isRequired,
  singleRedFlag: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  singleRedFlag: state.singleRedFlag
});

export default connect(mapStateToProps, { getSingleRedflag })(RedFlag);
