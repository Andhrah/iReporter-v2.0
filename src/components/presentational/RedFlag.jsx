import React from 'react';
import PropTypes from 'prop-types';
import Image from './Image';

const RedFlag = ({ redFlag }) => (
  <div className="wrapper">
    <div className="first-incident-container">

      <div className="jumbotron">
        <div className="title-container">
          <h2>
            Red-flag Report
          </h2>
        </div>
      </div>

      <div className="incident-container" id="single-red-flag">
        <div className="incident-image-lg" id="image-container">
          <Image src="http://lorempixel.com/400/400" alt="red-flag incident" />
        </div>

        <div className="author" id="created-by">
          <em id="created-on">
            {new Date(redFlag && redFlag.created_on).toDateString()}
          </em>
        </div>

        <div className="location">
          <p id="location-paragraph">
            Location:
            <span>
              <em id="lacation-span-em">{redFlag && redFlag.names_involved[0]}</em>
            </span>
          </p>
        </div>
        <iframe allowFullScreen src={`https://maps.google.com/maps?q=${redFlag.names_involved[0]}&output=embed&hl=en;z=20`} frameBorder="0" title={redFlag.names_involved[0]} className="map" />

        <div className="status">
          <p>
            Status:
            {' '}
            <span>
              <em id="statusEm">{redFlag && redFlag.status}</em>
            </span>
          </p>
        </div>

        <div className="comment">
          <h4 id="corruption-methods">{redFlag && redFlag.corruption_methods.join(', ')}</h4>
          <p id="corruption-date">{`corruption Date/Period: ${new Date(redFlag.created_on).toDateString()}`}</p>
          <p id="main-comment">
            {redFlag && redFlag.comment}
          </p>
        </div>

        {/* ------------------------------------------ */}
        {/* The Modal */}
        {/* ------------------------------------------ */}
        {/* <div id="myModal" className="modal"> */}
        {/* Modal content */}
        {/* <div className="modal-content">
      <span className="close">×</span>
      <p>
        Please Note: You can only edit the
        <strong>Location and Comment</strong>
        {' '}
        of a record.
      </p> */}
        {/* <div className="btn">
        What do you wish to edit?
        <button type="button" className="edit" id="edit-btn-comment">Edit Comment</button>
        <button type="button" className="edit" id="edit-btn-location">Edit Location</button>
      </div>
    </div>
  </div> */}

        {/* ---------------------- */}
        {/* Comment modal */}
        {/* ---------------------- */}
        {/* <div id="commentModal" className="modal"> */}
        {/* Modal content */}
        {/* <div className="modal-content">
      <span id="close" className="close">×</span>
      <textarea name id="comment" cols={50} rows={10} placeholder="Enter text here" defaultValue="" />
      <div className="btn">
        <button type="button" className="edit" id="save-btn-comment">Save Comment</button>
      </div>
    </div>
  </div> */}

        {/* ------------------------- */}
        {/* Location modal */}
        {/* ------------------------- */}
        {/* <div id="locationModal" className="modal"> */}
        {/* Modal content */}
        {/* <div className="modal-content">
      <span id="close-location-modal" className="close">×</span>
      <p>
        Location should follow this format
        {' '}
        <strong>eg. 6.605874, 3.349149</strong>
      </p>
      <div>
        <p className="error" id="location-error" />
        <label htmlFor="location">Location</label>
        <input name="location" id="location" type="text" />
      </div>
      <div className="btn">
        <button type="button" className="edit" id="save-btn-location">Save Location</button>
      </div>
    </div>
  </div> */}

        {/* ------------------------- */}
        {/* Red-Flag Deletion modal */}
        {/* ------------------------- */}

        {/* <div id="deleteModal" className="modal"> */}
        {/* Modal content */}
        {/* <div className="modal-content">
      <span id="close-delete-modal" className="close">×</span>
      <p>
        <strong>
        Are you sure you want to
          {' '}
          <em>delete</em>
          {' '}
        this record?
        </strong>
      </p>
      <div className="btn">
        <button type="button" id="cancel">Cancel</button>
        <button type="button" id="delete-red-flag" className="delete">Delete</button>
      </div>
    </div>
  </div> */}

        {/* <div className="btn-container">
          <div className="btn"><button type="button" className="edit" id="edit-btn">Edit</button></div>
          <div className="btn"><button type="button" id="delete" className="delete">Delete</button></div>
        </div> */}

      </div>
    </div>
  </div>
);

RedFlag.propTypes = {
  redFlag: PropTypes.object.isRequired
};

export default RedFlag;
