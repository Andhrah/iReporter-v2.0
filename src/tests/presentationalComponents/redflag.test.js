import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import RedFlag from '../../components/presentational/RedFlag';

const props = {
  redFlag: {
    id: 20,
    created_on: '2019-01-29T00:00:00.000Z',
    created_by: 1,
    corruption_methods: [
      'Bribery',
      ' Abuse of power'
    ],
    entity_involved: [
      'custom Officer',
      ' police'
    ],
    location: '6.605874, 3.349149',
    corruption_date: '2019-01-22T00:00:00.000Z',
    names_involved: [
      'Jane doe',
      ' John Bull'
    ],
    status: 'Under Investigation',
    images: [
      'image1',
      ' image2'
    ],
    videos: [
      'video1',
      ' video2'
    ],
    comment: 'Political corruption is a persistent phenomenon in Nigeria.'
  },
};

describe('RedFlag test suite', () => {
  it('renders correctly', () => {
    const wrapper = mount(
      <MemoryRouter>
        <RedFlag {...props} />
      </MemoryRouter>
    );
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });
});
