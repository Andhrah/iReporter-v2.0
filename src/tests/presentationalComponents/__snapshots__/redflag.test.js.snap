// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RedFlag test suite renders correctly 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "d0v8pz",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "d0v8pz",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <RedFlag
      redFlag={
        Object {
          "comment": "Political corruption is a persistent phenomenon in Nigeria.",
          "corruption_date": "2019-01-22T00:00:00.000Z",
          "corruption_methods": Array [
            "Bribery",
            " Abuse of power",
          ],
          "created_by": 1,
          "created_on": "2019-01-29T00:00:00.000Z",
          "entity_involved": Array [
            "custom Officer",
            " police",
          ],
          "id": 20,
          "images": Array [
            "image1",
            " image2",
          ],
          "location": "6.605874, 3.349149",
          "names_involved": Array [
            "Jane doe",
            " John Bull",
          ],
          "status": "Under Investigation",
          "videos": Array [
            "video1",
            " video2",
          ],
        }
      }
    >
      <div
        className="wrapper"
      >
        <div
          className="first-incident-container"
        >
          <div
            className="jumbotron"
          >
            <div
              className="title-container"
            >
              <h2>
                Red-flag Report
              </h2>
            </div>
          </div>
          <div
            className="incident-container"
            id="single-red-flag"
          >
            <div
              className="incident-image-lg"
              id="image-container"
            >
              <Image
                alt="red-flag incident"
                src="http://lorempixel.com/400/400"
              >
                <img
                  alt="red-flag incident"
                  className="red-flag-image"
                  src="http://lorempixel.com/400/400"
                />
              </Image>
            </div>
            <div
              className="author"
              id="created-by"
            >
              <em
                id="created-on"
              >
                Tue Jan 29 2019
              </em>
            </div>
            <div
              className="location"
            >
              <p
                id="location-paragraph"
              >
                Location:
                <span>
                  <em
                    id="lacation-span-em"
                  >
                    Jane doe
                  </em>
                </span>
              </p>
            </div>
            <iframe
              allowFullScreen={true}
              className="map"
              frameBorder="0"
              src="https://maps.google.com/maps?q=Jane doe&output=embed&hl=en;z=20"
              title="Jane doe"
            />
            <div
              className="status"
            >
              <p>
                Status:
                 
                <span>
                  <em
                    id="statusEm"
                  >
                    Under Investigation
                  </em>
                </span>
              </p>
            </div>
            <div
              className="comment"
            >
              <h4
                id="corruption-methods"
              >
                Bribery,  Abuse of power
              </h4>
              <p
                id="corruption-date"
              >
                corruption Date/Period: Tue Jan 29 2019
              </p>
              <p
                id="main-comment"
              >
                Political corruption is a persistent phenomenon in Nigeria.
              </p>
            </div>
          </div>
        </div>
      </div>
    </RedFlag>
  </Router>
</MemoryRouter>
`;
